### PR TITLE
buildscript cleanup

### PR DESCRIPTION
**Pull request by ethanmdavidson**
- use plugin dsl
- use shadowJar task instead of custom plugin task
- more consistent formatting
- upgrade kotlinx-coroutines to match upgraded kotlin (1.2.1 is companion to 1.3.31)

edit: also got integrations tests working in travis

### CODE REVIEW

**Comment by sanity** [comment]
Seeing some build errors in Travis, should this have worked?  https://travis-ci.org/kwebio/core/builds/537665644?utm_source=github_status&utm_medium=notification

**Comment by ethanmdavidson** [comment]
I didn't realize how PRs worked, so I kept working on master after creating it. It was supposed to just be the buildscript cleanup stuff.

Anyway, I got the tests working by specifying the ChromeDriver, but then I was trying to get the [generic driver](https://bonigarcia.github.io/selenium-jupiter/#generic-driver) feature working but it wasn't cooperating because you need to [explicitly pass the properties](https://stackoverflow.com/questions/42492742/how-to-pass-command-line-arguments-to-tests-with-gradle-test) through to the gradle task.

long story short: it should be working now

**Comment by sanity** [comment]
Oh, yes - it does appear to be working now with travis-ci, that's great!  I'll try to add some more tests.
